joplin/pages/service_page/tests.py

```diff
@@ -131,3 +112,74 @@ def test_create_service_page_with_dynamic_content_list():
 def test_kitchen_sink():
     page = fixtures.kitchen_sink()
     assert isinstance(page, ServicePage)
+
+
+@pytest.mark.django_db
+def test_create_service_page_with_step_with_1_location():
+    page = fixtures.step_with_1_location()
+    expected_steps = components.step_with_1_location()
+
+    assert isinstance(page, ServicePage)
+    assert page.title == "Service Page with 1 location step"
+    assert page.slug == "service-page-with-1-location-step"
+    for i, step in enumerate(page.steps.stream_data):
+        assert step["type"] == expected_steps[i]["type"]
+        assert step["value"] == expected_steps[i]["value"]
+
+
+@pytest.mark.django_db
+def test_create_service_page_with_step_with_2_locations():
+    page = fixtures.step_with_2_locations()
+    expected_steps = components.step_with_2_locations()
+
+    assert isinstance(page, ServicePage)
+    assert page.title == "Service Page with 2 locations step"
+    assert page.slug == "service-page-with-2-locations-step"
+    for i, step in enumerate(page.steps.stream_data):
+        assert step["type"] == expected_steps[i]["type"]
+        assert step["value"] == expected_steps[i]["value"]
+
+
+@pytest.mark.django_db
+def test_import_step_with_1_already_existing_location(remote_staging_preview_url, test_api_url, test_api_jwt_token):
+    location_page = location_page_fixtures.title()
+    url = f'{remote_staging_preview_url}/services/UGFnZVJldmlzaW9uTm9kZTo1OA==?CMS_API={test_api_url}'
+    page = PageImporter(url, test_api_jwt_token).fetch_page_data().create_page()
+    assert isinstance(page, ServicePage)
+    assert page.live
+    # The importer should assign the step's location to our matching location_page.
+    assert page.steps.stream_data[0]["value"]["locations"][0] == location_page.pk
+
+
+@pytest.mark.django_db
+def test_import_step_with_1_new_location(remote_staging_preview_url, test_api_url, test_api_jwt_token):
+    url = f'{remote_staging_preview_url}/services/UGFnZVJldmlzaW9uTm9kZTo1OA==?CMS_API={test_api_url}'
+    page = PageImporter(url, test_api_jwt_token).fetch_page_data().create_page()
+    assert isinstance(page, ServicePage)
+    assert not page.live
```

Comment:
thinking about what goes live for full import

current logic:
Location pages:
* all will go live (related services will be set to a placeholder)
Service pages:
* should all go live because location pages [will have already been imported](https://github.com/cityofaustin/joplin/pull/691/commits/63bc25c95758d517e465cd9a4b8b5161ff4e0ff2)

I think this logic should be fine for MVP, we have [20 related service instances in prod](https://joplin.herokuapp.com/api/graphiql#query=%7B%0A%20%20allLocationPages%20%7B%0A%20%20%20%20edges%20%7B%0A%20%20%20%20%20%20node%20%7B%0A%20%20%20%20%20%20%20%20relatedServices%20%7B%0A%20%20%20%20%20%20%20%20%20%20edges%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20node%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20hoursExceptions%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20page%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20title%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) so it should be a pretty quick manual process to update them from the placeholder